docs/source/configuration/authn-jwt.mdx

```diff
@@ -145,54 +146,30 @@ The default value is `15s` (fifteen seconds), which is suitable for almost all u
+The authentication plugin performs validation in the `RouterService`, which is the _first_ service that executes in the Apollo Router's [request pipeline](../customizations/overview/#how-customizations-work). This means you can build [router customizations](../customizations/overview/) (usually [Rhai scripts](../customizations/rhai/)) that interact with claims by hooking into any _other_ service.
 
-Because claims are added to the context, you can define custom logic for handling each request based on the details of its claims. You can define this logic as a Rhai script or native Rust plugin (for more on these, see [Customizations for the Apollo Router](../customizations/overview/)).
+> **Important:** You _cannot_ build a router customization that interacts with an incoming JWT _before_ the authentication plugin validates it. Plugins that hook into the `RouterService` execute in an undefined order and might even execute in parallel.
 
-### Example: Forwarding claims to subgraphs
-
-Below is an example Rhai script that forwards a JWT's claims to individual subgraphs via HTTP headers (one header for each claim). This enables each subgraph to define logic to handle (or potentially reject) incoming requests based on claim details.
-
-> [Learn more about creating Rhai scripts.](../customizations/rhai/)
-
-<ExpansionPanel title="Click to expand">
-
-```rhai title="claims_forwarding.rhai"
-fn subgraph_service(service, subgraph) {
-  let f = |request| {
-    // Note: APOLLO_AUTHENTICATION_JWT_CLAIMS is a Rhai-scope constant with
-    // value `apollo_authentication::JWT::claims`
-    let claims = request.context[APOLLO_AUTHENTICATION_JWT_CLAIMS];
-    if claims ==() {
-      throw #{
-        status: 401
-      };
-    }
-    // Add each claim key-value pair as a separate HTTP header
-    for key in claims.keys() {
-      request.subgraph.headers[key] = claims[key];
-    }
-  };
+Below are two example [Rhai script](../customizations/rhai/) customizations that demonstrate actions the router can perform based on a request's claims.
 
-  service.map_request(f);
-}
-```
-
-</ExpansionPanel>
+### Example: Validating claims
 
-### Example: Throwing errors for invalid claims
+Below is an example [Rhai script](../customizations/rhai/) that checks the details of different JWT claims and throws different errors for different invalid claim details.
 
-Below is an example Rhai script that throws distinct errors for different invalid JWT claim details.
-
-> [Learn more about creating Rhai scripts.](../customizations/rhai/)
+> This script hooks into the router's `SupergraphService`, which executes before the router begins generating the query plan for an operation. [Learn more about router services.](../customizations/overview#service-descriptions)
 
 <ExpansionPanel title="Click to expand">
 
 ```rhai title="claims_validation.rhai"
+fn supergraph_service(service) {
```

Comment:
Should the other example (below) and all _other_ Rhai examples in our docs use this pattern? I reckon we should standardize on _some_ pattern to simplify Rhai for folks

Reply:
> Should the other example (below) and all _other_ Rhai examples in our docs use this pattern? I reckon we should standardize on _some_ pattern to simplify Rhai for folks

Some people like function pointers, some people like closures... I will note that there is a conflict:
 - function pointers are higher performance (although, it's probably in the land of "marginal gains")
 - closures are more "javascript developer friendly" (I think)

Rather than favour one style over the other, some examples use function pointers and some use closures to try and provide illustrations for both mechanisms and cover all tastes.

Reply:
okay! Happy to keep both, and I've clarified the distinction in the JWT doc